types/modules/producer.d.ts

```diff
@@ -0,0 +1,87 @@
+import { Connection } from './connection';
+import { ProducerHooks } from './hooks';
+import type amqp = require('amqplib');
+import pDefer = require('p-defer');
+
+declare class ProducerError extends Error {
+  constructor(error: { name: string; message: string });
+}
+
+interface PublishOptions extends amqp.Options.Publish {
+  routingKey?: string;
+  rpc?: boolean;
+}
+
+declare class Producer {
+  constructor(connection: Connection);
+  hooks: ProducerHooks;
+  /**
+   * Map of rpc queues
+   *
+   * [queue: string] -> [correlationId: string] -> {responsePromise, timeoutId}
+   */
+  amqpRPCQueues: Record<
+    string,
+    Record<string, { responsePromise: pDefer.DeferredPromise<unknown>; timeoutId: NodeJS.Timeout }>
+  >;
+  private _connection: Connection;
+  set connection(value: Connection);
```

Comment:
don't think we should allow setting connection .. 
```suggestion
  private set connection(value: Connection);
```

Reply:
This is actually used internally to replace the connection so it's not really private.
This should be cleaned as part of https://github.com/bringg/node-arnavmq/issues/68 since the replaced collection is always the same instance anyways and it doesn't make sense there, but on it's singleton form that's how it is.

https://github.com/bringg/node-arnavmq/blob/35a84612f08ee64bfd4d45928c67bca7350c14cd/src/modules/arnavmq.js#L49

Reply:
I don't expect someone to do `arnavmq.producer.connection = ...`, and if so I would consider it as bad practice.

I think it should be ok to mark it as private, and if once we move to TypeScript, have the connection be accepted at the constructor and don't expose it.